File: ios/Plugin/AllFilesAccessPlugin.m
#import <Foundation/Foundation.h>
#import <Capacitor/Capacitor.h>

// Define the plugin using the CAP_PLUGIN Macro, and
// each method the plugin supports using the CAP_PLUGIN_METHOD macro.
CAP_PLUGIN(AllFilesAccessPlugin, "AllFilesAccess",
           CAP_PLUGIN_METHOD(echo, CAPPluginReturnPromise);
)

File: ios/Plugin/AllFilesAccessPlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(AllFilesAccessPlugin)
public class AllFilesAccessPlugin: CAPPlugin {
    private let implementation = AllFilesAccess()

    @objc func echo(_ call: CAPPluginCall) {
        let value = call.getString("value") ?? ""
        call.resolve([
            "value": implementation.echo(value)
        ])
    }
}
